client/landing/stepper/hooks/use-get-domain-price-override.ts

```diff
@@ -0,0 +1,58 @@
+import { keepPreviousData, useQuery } from '@tanstack/react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+import { DomainPriceOverrideOptions } from '../declarative-flow/internals/steps-repository/domain-transfer-domains/use-price-override';
+import type { DomainLockResponse } from './use-is-domain-code-valid';
+
+const VERSION = 1;
+const HUNDRED_YEAR_DOMAIN_VENDOR = '100-year-domains';
+
+type DomainPriceOverrideResponse = Pick<
+	DomainLockResponse,
+	'raw_price' | 'sale_cost' | 'currency_code'
```

Comment:
For the price override, we only care about these properties rather than all the others returned by the availability endpoint - we could also use the `/price` endpoint, but we opted for the `/is-available` one to standardize it and make both flows/all branches work the same way.